timbermill-java/timbermill-local/src/main/java/com/datorama/oss/timbermill/ElasticsearchClient.java

```diff
@@ -942,11 +938,170 @@ long countByName(String name, String env) throws IOException {
+    }
+
+    private List<String> getIndicesToBeMoved(String alias) throws IOException {
+        InputStream content = null;
+        try {
+            Set<String> indexNotToMerge = getAliases(alias + "*").keySet();
+            Request request = new Request("GET", "/_cat/indices/" + alias + "*?format=json");
+            Response response = client.getLowLevelClient().performRequest(request);
+            content = response.getEntity().getContent();
+            String json = IOUtils.toString(content);
+            JsonArray asJsonObject = new JsonParser().parse(json).getAsJsonArray();
+            List<String> indicesToMove = new ArrayList<>();
+            asJsonObject.forEach(jsonElement -> {
+                JsonObject jsonObject = jsonElement.getAsJsonObject();
+                String storeSize = jsonObject.get("store.size").getAsString();
+                String index = jsonObject.get("index").getAsString();
+                if (indexNotToMerge.contains(index)) {
+                    return;
+                }
+                if (isSizeLowerThen(storeSize, (long) (maxIndexSizeInGB * 0.1))) { // 10% of max index size
+                    indicesToMove.add(index);
+                    LOG.info("Index {} has a size of {}. Will be moved to another index", index, storeSize);
+                }
+            });
+            LOG.info("Number of indices to be moved: {}", indicesToMove.size());
+            return indicesToMove;
+        } catch (Exception e) {
+            LOG.error("Failed to get indices to be moved", e);
+            return Collections.emptyList();
+        } finally {
+            if (content != null) {
+                content.close();
+            }
+        }
+    }
+
+    private boolean isSizeLowerThen(String storeSize, long maxIndexSizeInGB) {
+        String sizeUnit = storeSize.substring(storeSize.length() - 2);
+
+        if (sizeUnit.equalsIgnoreCase("kb") || sizeUnit.equalsIgnoreCase("mb")) {
+            return true;
+        }
+
+        if (storeSize.length() < 2) {
+            LOG.info("Store size {} is not in the right format", storeSize);
+            return false;
+        }
+
+        String sizeStr = storeSize.substring(0, storeSize.length() - 2);
+        double size = Double.parseDouble(sizeStr);
+        return size < maxIndexSizeInGB;
+    }
+
+    private void moveIndexAnotherIndex(String sourceIndex, String destIndex) {
+        LOG.info("Moving index {} to {}", sourceIndex, destIndex);
+        ReindexRequest reindexRequest = new ReindexRequest();
+        reindexRequest.setConflicts("proceed");
+        reindexRequest.setSourceIndices(sourceIndex);
+        reindexRequest.setDestIndex(destIndex);
+        reindexRequest.setDestOpType("create");
+        reindexRequest.setSlices(10);
+        reindexRequest.setTimeout(TimeValue.timeValueMinutes(10));
```

Comment:
better to extract to property, we might need to enlarge timeouts for big re-index operations.

Reply:
there are a lot of params as it is i don't want to add more